### DISCUSSION TITLE
Setting up next-auth(email, google, apple) with hasura

### DISCUSSION THREAD

**Discussion by Manubi** [nextauthjs/next-auth#1297] [Help]
Hey everyone!

It's my first time working with next-auth so please be patient. I do like it so far! 💯 
I will go threw all the tutorials I find until I feel confident in making the switch, but before I'd like to get some questions out of the way. Thanks!

**Status quo:**
So far, we are using firebase-auth with JWT to connect to an existing hasura graphql backend that uses postgres. But we are not happy with the firebase serverless onCreate hook that takes always too long when a user registers.
I was able to add next-auth on the frontend with googleOauth but without database access. So I only have the google provider in my `[...nextAuth].ts` and login/logout works on the frontend.

**What I am trying to achieve:**
I want to add next-auth to our existing project and want to use email (magic link), google and apple oauth with JWT to connect to our hasura/postgres backend.

**What are the questions:**

1. The first step would probably be adding the model to the database. But the thing is we already have a `user` table (not `users` as used by next-auth) and we would like to just add the missing columns in order to not have to rewrite all the graphql queries in the frontend. Is that possible or do I need to change the table name to `user` to use next-auth? If it's possible with an adapter can you lead me to a tutorial where I can learn about it? Thanks!
2. As I've seen from the [models section](https://next-auth.js.org/schemas/models) that `session` is only needed if you don't use JWT authentication. So I guess we won't have to add the `session` table, right?
3. As we are using uuid as our primary identifier (`id` type uuid) what's the best way to change it from integer to uuid? **Update:** _You can add properties to the schema and map them to any database column names you wish, but you should not change the base properties or types defined in the example schema._ Means probably no it's not possible?
4. Will this "more work" make sense in the long term if we want to stay with next-auth or does it just complicate things and I should just change everything to the standard version of next-auth? As we will want to use 2FA as soon as it gets added...

Thanks for your patience and help!

cheers
Manuel

**Comment by Manubi** [comment]
Someone? 😄

**Comment by Manubi** [comment]
Someone? 😄

**Comment by arnaudjnn** [comment]
Hello @Manubi 

Here is my starter example: https://github.com/arnaudjnn/nextjs-hasura-starter

1. Best would be to use `users` table name
2. You don't need a session table but you can add as many attributes as you want in jwt callback.
3. You can change from id to uuid in Hasura right?
4. No opinion
